pkg/ruler/scheduler.go

```diff
@@ -287,7 +309,7 @@ func (s *scheduler) workItemDone(i workItem) {
 		level.Debug(util.Logger).Log("msg", "scheduler: stopping item", "user_id", i.userID, "group", i.groupName, "found", found, "len", len(currentRules))
 		return
 	}
-	next := i.Defer(s.evaluationInterval, i.groupName, currentRules)
+	next := i.Defer(s.evaluationInterval)
```

Comment:
I can't see how we are now ensuring that, after a rules update, the new rules are used at this point.
This was issue #482

Reply:
My understanding of the code was that by checking `i.generation < config.generation` and returning if the workItem is old would mean that old rules would never make it to the Defer statement. Is that not correct or am I missing some other edge case?

Reply:
ah, hadn't seen that go in with #827